packages/lib/services/searchengine/SearchEngineUtils.ts

```diff
@@ -49,9 +50,24 @@ export default class SearchEngineUtils {
 		// issue: https://discourse.joplinapp.org/t/how-to-recover-corrupted-database/9367
 		if (noteIds.length !== notes.length) {
 			// remove null objects
-			return sortedNotes.filter(n => n);
+      const results2:any = [];
```

Comment:
Appreciate it might not be possible to have the exact type, but at least you can say it's `any[]` since it's an array.

Reply:
I made the following changes to my commit:
1. Reformatted the code properly
2. the for loop runs only when the showCompletedTodos is toggled.
3. changed const results2:any = [ ]; to const results2:any[ ] = [ ]; so that it supports all elements(not sure if you desired this)

Are the changes sufficient or do you wish any more changes?